### PR TITLE
Accession import identifier handling, refs #12043

### PR DESCRIPTION
**Pull request by mcantelon**
Changed CSV accession import so if an accession number isn't specified the row
is skipped, unless the assign-id option is used when running the import task.

Fixed issue with generating identifiers based on masks/counters from the CLI.

### CODE REVIEW

File: lib/task/import/csvAccessionImportTask.class.php [hunk 1]
```diff
@@ -102,7 +108,8 @@ public function execute($arguments = array(), $options = array())
         'acquisitionTypes'   => $termData['acquisitionTypes'],
         'resourceTypes'      => $termData['resourceTypes'],
         'processingStatus'   => $termData['processingStatus'],
-        'processingPriority' => $termData['processingPriority']
+        'processingPriority' => $termData['processingPriority'],
+        'assign-id'          => $options['assign-id']
```
Comment (jraddaoui) [inline lib/task/import/csvAccessionImportTask.class.php]: After getting the value from the options, we should camelCase the var like the others are.

File: lib/task/import/csvAccessionImportTask.class.php [hunk 2]
```diff
@@ -192,23 +199,33 @@ public function execute($arguments = array(), $options = array())
         $result = $statement->fetch(PDO::FETCH_OBJ);
         if ($result)
         {
-          print 'Found '. $result->id ."\n";
+          print $self->logError(sprintf('Found accession ID %d', $result->id));
```
Comment (jraddaoui) [inline lib/task/import/csvAccessionImportTask.class.php:204]: Nice.

File: lib/task/import/csvAccessionImportTask.class.php [hunk 3]
```diff
@@ -192,23 +199,33 @@ public function execute($arguments = array(), $options = array())
...
           $self->object = QubitAccession::getById($result->id);
         }
-        else
+        else if (!empty($accessionNumber))
         {
-          $self->object = false;
```
Comment (jraddaoui) [inline lib/task/import/csvAccessionImportTask.class.php:200]: hehe

File: plugins/qtAccessionPlugin/lib/model/QubitAccession.php [hunk 4]
```diff
@@ -103,7 +103,9 @@ public static function nextAvailableIdentifier()
       $con->beginTransaction();
 
       // Determine what should be the next identifier
-      $identifier = Qubit::generateIdentifierFromCounterAndMask(self::nextAccessionNumber(), sfConfig::get('app_accession_mask'));
+      $setting = QubitSetting::getByName('accession_mask');
+      $mask = $setting->getValue(array('sourceCulture' => true));
+      $identifier = Qubit::generateIdentifierFromCounterAndMask(self::nextAccessionNumber(), $mask);
```
Comment (jraddaoui) [inline plugins/qtAccessionPlugin/lib/model/QubitAccession.php]: Why do you need to do this? `sfConfig` should be populated with the settings in CLI tasks if they extend `arBaseTask` and I believe `csvAccessionImportTask` inherits from it through `csvImportBaseTask`.

**Review by jraddaoui — COMMENTED** [review]
Looks great @mcantelon. A minor nitpick and a question.

File: plugins/qtAccessionPlugin/lib/model/QubitAccession.php (hunk 4 above)
Comment (jraddaoui) [reply plugins/qtAccessionPlugin/lib/model/QubitAccession.php]: Oh, I see. We're not calling the `parent::execute` from `csvAccessionImportTask`, like we do in:

https://github.com/artefactual/atom/blob/6ad917404acd4f2a8c53f90c034b99d4fb2312e1/lib/task/import/csvImportTask.class.php#L138

If you could add it and keep using sfConfig in here to maintain consistency, that would be great.

**Review by jraddaoui — CHANGES_REQUESTED** [review]
A minor change requested in the end ;)

File: plugins/qtAccessionPlugin/lib/model/QubitAccession.php (hunk 4 above)
Comment (mcantelon) [reply plugins/qtAccessionPlugin/lib/model/QubitAccession.php]: Ah! Good to know! I'll change that.

File: lib/task/import/csvAccessionImportTask.class.php (hunk 1 above)
Comment (mcantelon) [reply lib/task/import/csvAccessionImportTask.class.php]: Good call.

**Review by jraddaoui — APPROVED** [review]
Great, thanks @mcantelon!

**Pull request merged**